src/StorageFlattening.cpp

```diff
@@ -535,13 +535,9 @@ class FlattenDimensions : public IRMutator {
             Interval loop_bounds = Interval(expanded_min, simplify(expanded_min + expanded_extent - 1));
             it->loop_vars.push(op->name, loop_bounds);
         }
-        bool old_in_gpu = in_gpu;
-        if (op->for_type == ForType::GPUBlock ||
-            op->for_type == ForType::GPUThread) {
-            in_gpu = true;
-        }
+
+        ScopedValue<bool> old_in_gpu(in_gpu, in_gpu || is_gpu(op->for_type));
```

Comment:
fyi: I'm liking the way that Slinky does this, i.e. `auto saved = make_save_thing<bool>(new-value);` as I find it more readable. Not proposing we do a mass change but would be nice to think about.